ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/SendRequest.java

```diff
@@ -0,0 +1,46 @@
+package com.mozilla.telemetry.contextualservices;
+
+import com.mozilla.telemetry.ingestion.core.Constant.Attribute;
+import com.mozilla.telemetry.transforms.FailureMessage;
+import com.mozilla.telemetry.transforms.PubsubConstraints;
+import java.io.UncheckedIOException;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.WithFailures.ExceptionElement;
+import org.apache.beam.sdk.transforms.WithFailures.Result;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TypeDescriptor;
+
+/**
+ * Send GET requests to reporting.
+ */
+public class SendRequest extends
+    PTransform<PCollection<PubsubMessage>, Result<PCollection<PubsubMessage>, PubsubMessage>> {
+
+  public static SendRequest of() {
+    return new SendRequest();
+  }
+
+  @Override
+  public Result<PCollection<PubsubMessage>, PubsubMessage> expand(
+      PCollection<PubsubMessage> messages) {
+    return messages.apply(
+        MapElements.into(TypeDescriptor.of(PubsubMessage.class)).via((PubsubMessage message) -> {
+          message = PubsubConstraints.ensureNonNull(message);
+
+          // TODO: HTTP GET
```

Comment:
I did a little review of Java HTTP clients. If we could use Java 11, I'd recommend using the new built-in HttpClient, but we're not there yet.

https://square.github.io/okhttp/ looks to be pretty widely used and the simplest interface.

The more conservative choice would be Apache HttpComponents classic, which is probably reasonably easy to use: http://hc.apache.org/httpcomponents-client-5.0.x/index.html